packages/beeq/src/components/date-picker/bq-date-picker.tsx

```diff
@@ -239,6 +246,22 @@ export class BqDatePicker {
     this.open = ev.detail.open;
   }
 
+  @Listen('mousedown', { target: 'window', capture: true })
+  async handleMouseClick(event: MouseEvent) {
+    const { open, datePickerElement, type, hasRangeEnd } = this;
+
+    if (!open || !datePickerElement || type !== 'range' || event.button !== 0) return;
+
+    const { top, bottom, left, right } = datePickerElement.getBoundingClientRect();
+    const { clientY, clientX } = event;
+
+    const isOutside = clientY < top || clientY > bottom || clientX < left || clientX > right;
+
+    if (isOutside && !hasRangeEnd) {
+      this.setTentative = this.setTentative === undefined ? null : undefined;
+    }
+  }
```

Comment:
We could simplify this by using the `isEventTargetChildOfElement` utility from `packages/beeq/src/shared/utils/event.ts`, and I will recommend listening for the `click` event instead of `mousedown`:

```suggestion
  @Listen('click', { target: 'window', capture: true })
  async handleMouseClick(event: MouseEvent) {
    const { open, datePickerElement, type, hasRangeEnd } = this;
    if (!open || !datePickerElement || type !== 'range' || event.button !== 0) return;

    if (!isEventTargetChildOfElement(event, this.el) && !hasRangeEnd) {
      this.tentative = undefined;
    }
  }
```

Reply:
By assigning `undefined`, you can see that it is enough to clear the `tentative` value.

Reply:
Keep consistency when using the `event` argument. In this file, all the methods use `ev` instead of `event`.